clicommand/agent_start.go

```diff
@@ -731,6 +733,9 @@ var AgentStartCommand = cli.Command{
 				l.Info("Registering agent %d of %d with Buildkite...", i, cfg.Spawn)
 			}
 
+			// Handle per-spawn name interpolation, replacing %spawn with the spawn index
+			registerReq.Name = strings.ReplaceAll(cfg.Name, "%spawn", strconv.Itoa(i))
```

Comment:
Interesting… I guess this mutates the shared `api.AgentRegisterRequest` on each `for` iteration.
We could jump through some hoops to copy the request, but there's probably no good reason, unless if/when we decide to do the registrations in parallel, at which point I'm sure we'll scrutinise for things like this.
So… carry on :)

Reply:
That's right, yeah. I didn't see a reason to avoid this right now, but it wouldn't be too hard to swap it to create an instance on each iteration either.

I agree that if we do parallel registrations we will probably instead be passing the config to each worker instance and having it be responsible for the registration, which will avoid the reuse.